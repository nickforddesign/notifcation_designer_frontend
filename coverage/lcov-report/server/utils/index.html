<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for server/utils</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> server/utils
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">58.97% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>115/195</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">17.78% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>8/45</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">69.44% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>25/36</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">59.07% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>114/193</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="58.97" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 58%;"></div><div class="cover-empty" style="width:42%;"></div></div></td>
	<td data-value="58.97" class="pct medium">58.97%</td>
	<td data-value="195" class="abs medium">115/195</td>
	<td data-value="17.78" class="pct low">17.78%</td>
	<td data-value="45" class="abs low">8/45</td>
	<td data-value="69.44" class="pct medium">69.44%</td>
	<td data-value="36" class="abs medium">25/36</td>
	<td data-value="59.07" class="pct medium">59.07%</td>
	<td data-value="193" class="abs medium">114/193</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Sep 08 2017 12:51:44 GMT-0400 (EDT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
