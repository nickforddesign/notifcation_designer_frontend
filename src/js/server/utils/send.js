const email = require('sendmail')()
const utils = require('./index')
const inline = require('../inline')

module.exports = class Email {
  constructor(options) {
    return this.send(options)
  }
  async send(options) {
    const html = await inline(options.body)
    try {
      return utils.asyncCallback(email, {
        from: options.from,
        to: options.to,
        subject: options.subject,
        html
      })
    } catch (error) {
      console.warn(error)
    }
  }
}
