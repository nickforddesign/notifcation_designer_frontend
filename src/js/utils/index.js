// import _ from 'lodash'
// just chill

export const sleep = (ms) => {
  return new Promise(resolve => setTimeout(resolve, ms))
}

export const pathToData = (path, files) => {
  // const _path = path.split('/').join('/data/').split('/').slice(2).join('/')
  const data = searchTree(path.split('/').slice(1), files)
  return data
  // const data = _.get(files, _path)
  // console.log(data)
  // console.log(path, JSON.parse(JSON.stringify(files)))
}

function searchTree (path_array, files) {
  if (path_array.length) {
    const match = files.find(file => {
      return path_array[0] === file.name
    })
    path_array = path_array.slice(1)
    if (path_array.length && match) {
      return searchTree(path_array, match.data)
    } else {
      return match
    }
  }
}
/*
export const processFolder = (object) => {
  let output = []
  for (let key in object) {
    let file = {}
    file.name = key
    if (typeof object[key] === 'string') {
      file.type = 'file'
      file.data = object[key]
    } else {
      file.type = 'folder'
      file.data = processFolder(object[key])
    }
    output.push(file)
  }
  return output
}
*/
