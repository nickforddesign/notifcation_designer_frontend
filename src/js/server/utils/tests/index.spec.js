import utils from '../index'
import glob from 'glob'

describe('utils.sleep', () => {
  it('should work with then', done => {
    utils.sleep(5)
      .then(() => {
        done()
      })
  })

  it('should work with await', async done => {
    await utils.sleep(5)
    done()
  })
})

describe('utils.Deferred', () => {
  it('should return a promise', () => {
    const deferred = new utils.Deferred()
    expect(deferred.promise)
      .toBeInstanceOf(Promise)
  })

  it('should resolve the promise', async () => {
    expect.assertions(1)
    const asyncFunc = (msg) => {
      const deferred = new utils.Deferred()
      utils.sleep(5)
        .then(() => {
          deferred.resolve(msg)
        })
      return deferred.promise
    }
    const msg = await asyncFunc('test')
    expect(msg)
      .toBe('test')
  })

  it('should reject the promise', () => {
    expect.assertions(1)
    const asyncFunc = (msg) => {
      const deferred = new utils.Deferred()
      deferred.reject(msg)
      return deferred.promise
    }

    asyncFunc('test')
      .catch(error => {
        expect(error)
          .toBe('test')
      })
  })
})

describe('utils.asyncCallback', () => {
  it('should turn a callback into async', async () => {
    expect.assertions(1)
    const callbackFunc = (param, callback) => {
      return callback(false, param)
    }
    const asyncFunc = (param) => {
      return utils.asyncCallback(callbackFunc, param)
    }
    const msg = await asyncFunc('test')
    expect(msg)
      .toBe('test')
  })
})

describe('utils.transformAsync', () => {
  it('should transform contents of promise', async () => {
    expect.assertions(1)
    const callbackFunc = (param, callback) => {
      return callback(false, param)
    }
    const asyncFunc = (param) => {
      return utils.transformAsync(
        utils.asyncCallback(callbackFunc, param),
        (array) => {
          return array.sort((a, b) => {
            return a > b
          })
        }
      )
    }
    const arr = await asyncFunc([3, 2, 1])
    expect(arr)
      .toEqual([1, 2, 3])
  })
})

describe('utils.mkdir', () => {
  it('should make a new directory asynchronously', async () => {
    expect.assertions(1)
    await utils.mkdir('./___testing___')
    const files = await utils.asyncCallback(glob, './___testing___')
    expect(files)
      .toEqual([
        './___testing___'
      ])
  })

  it('should make another directory asynchronously', async () => {
    expect.assertions(1)
    await utils.mkdir('./___testing___/test')
    const files = await utils.asyncCallback(glob, './___testing___/*')
    expect(files)
      .toEqual([
        './___testing___/test'
      ])
  })
})

describe('utils.writeFile', () => {
  it('should write a file asynchronously', async () => {
    await utils.writeFile('./___testing___/test/test.txt', 'testing')
    const files = await utils.asyncCallback(glob, './___testing___/test/*')
    expect(files)
      .toEqual([
        './___testing___/test/test.txt'
      ])
  })
})

describe('utils.appendFile', () => {
  it('should write a file asynchronously', async () => {
    await utils.appendFile('./___testing___/test/test.txt', 'testing')
    const files = await utils.asyncCallback(glob, './___testing___/test/*')
    expect(files)
      .toEqual([
        './___testing___/test/test.txt'
      ])
  })
})

describe('utils.readFile', () => {
  it('should read a file asynchronously', async () => {
    const txt = await utils.readFile('./___testing___/test/test.txt')
    expect(txt)
      .toBe('testingtesting')
  })
})

describe('utils.readdir', () => {
  it('should read a directory and return relative paths', async () => {
    const files = await utils.readdir('./___testing___')
    expect(files)
      .toEqual([
        'test'
      ])
  })
})

describe('utils.readdirPaths', () => {
  it('should read a directory and return full paths', async () => {
    const files = await utils.readdirPaths('./___testing___')
    expect(files)
      .toEqual([
        './___testing___/test'
      ])
  })
})

describe('utils.readdirRecursive', () => {
  it('should read a directory recursively, asynchronously', async () => {
    const files = await utils.readdirRecursive('./___testing___')
    expect(files)
      .toEqual([
        './___testing___/test',
        './___testing___/test/test.txt'
      ])
  })
})

describe('utils.pathsToTree', () => {
  it('should turn paths array into tree', async () => {
    const files = await utils.readdirRecursive('___testing___')
    const tree = await utils.pathsToTree(files)
    expect(tree)
      .toEqual({
        '___testing___': {
          'data': {
            'test': {
              'path': '___testing___/test',
              'type': 'folder',
              'data': {
                'test.txt': {
                  'path': '___testing___/test/test.txt',
                  'type': 'file'
                }
              }
            }
          }
        }
      }
    )
  })
})

describe('utils.treeToArray', () => {
  it('should convert the tree structure to nested arrays', async () => {
    const files = await utils.getFilesData('___testing___/')

    expect(files)
      .toEqual([
        {
          'data': [
            {
              'ext': 'txt',
              'name': 'test.txt',
              'path': 'test/test.txt',
              'type': 'file'
            }
          ],
          'name': 'test',
          'path': 'test',
          'type': 'folder'
        }
      ]
    )
  })
})

describe('utils.rmrf', () => {
  it('should remove a file asynchronously', async () => {
    await utils.rmrf('./___testing___/')
    const files = await utils.asyncCallback(glob, './___testing___/*')
    expect(files)
      .toEqual([])
  })
})
