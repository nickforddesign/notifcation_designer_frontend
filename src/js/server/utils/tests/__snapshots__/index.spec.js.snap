// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils.renderEmail should render email 1`] = `
Object {
  "html": "
  <style>body {
  font-family: \\"Avenir\\", Helvetica, Arial, sans-serif;
  background: #efefef; }

.email_html_body {
  background: #666666; }

.email_container {
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-weight: 200;
  font-size: 16px;
  color: #333;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  max-width: 550px;
  margin: 0px auto; }

.logo_container {
  padding: 20px; }

.logo {
  display: inline-block;
  vertical-align: top;
  max-width: 100px; }

.icon {
  margin: 20px 0 10px;
  max-width: 60px; }

.small {
  margin: 6px 0 6px;
  max-width: 24px; }

.divider {
  margin: 20px -20px;
  border-bottom: 2px solid #efefef; }

.email_body {
  padding: 20px 20px 10px 20px;
  background: #ffffff;
  -webkit-border-radius: 4px;
  border-radius: 4px;
  text-align: center; }

ul {
  margin: 0;
  padding: 10px 10px 10px 20px;
  font-size: 14px;
  list-style-type: square; }

li {
  margin: 8px;
  text-align: left;
  color: #6d6e71;
  mso-special-format: bullet; }

h1 {
  font-size: 24px;
  margin: 0;
  padding: 10px 10px 0 10px;
  color: #6d6e71;
  font-weight: 500; }

.negative {
  color: #e91e63; }

.positive {
  color: #56ba77; }

h2 {
  font-size: 22px;
  font-weight: 200;
  margin: 0px;
  padding: 20px;
  color: #6d6e71; }

h2 {
  font-size: 18px;
  font-weight: 500;
  margin: 0px;
  padding: 0px;
  color: #6d6e71; }

p {
  line-height: 22px;
  margin: 0px;
  padding: 10px;
  color: #6d6e71; }

p.small_text {
  font-size: 12px;
  line-height: 1.5; }

p.small_text.strong {
  font-weight: bold;
  width: 90%;
  margin: 0 auto; }

p.intro_text {
  font-size: 14px;
  line-height: 22px;
  margin-top: 10px; }

a {
  color: #0071bc; }

sup {
  font-size: 7px; }

.strong {
  font-weight: bold; }

.text-left {
  text-align: left; }

.text-center {
  text-align: center; }

.text-right {
  text-align: right; }

.email_footer {
  background: #efefef; }

.signature {
  font-size: 11px;
  line-height: 1.2;
  padding: 20px 0;
  color: #666;
  margin: 0px; }

.summary_table {
  margin-top: 20px;
  margin-bottom: 10px;
  border-spacing: 0;
  font-size: 12px;
  color: #6d6e71; }

.summary_cell {
  border-top: 1px solid #6d6e71; }

.activation_table {
  margin-top: 30px; }

.status {
  display: inline-block;
  padding: 4px 6px;
  -webkit-border-radius: 3px;
  border-radius: 3px;
  font-size: 12px;
  font-weight: 200; }

.status.pending {
  background-color: #007dc1;
  color: #ffffff; }

.status.processed {
  background-color: #56ba77;
  color: #ffffff; }

.status.error,
.status.failed,
.status.cancelled {
  background-color: #e91e63;
  color: #ffffff; }
</style>
  <!DOCTYPE html>
<html lang=\\"en\\">

<body class=\\"email_html_body\\">
<table border=\\"0\\" cellspacing=\\"0\\" cellpadding=\\"0\\" width=\\"100%\\" bgcolor=\\"#efefef\\">

  <!-- header -->
  <tr>
    <td bgcolor=\\"#efefef\\" align=\\"center\\">
      <div class=\\"logo_container\\">
        <a href=\\"http://localhost:8080\\">
          <img class=\\"logo\\" src=\\"http://localhost:8080/static/email_logo.png\\" title=\\"Rafi Payment\\" alt=\\"Rafi Payment\\" />
        </a>
      </div>
    </td>
  </tr>

  <!-- body -->
  <tr>
    <td bgcolor=\\"#efefef\\">
      <div class=\\"email_container\\">

        <div class=\\"email_body\\">
          
                <div>test123</div>
                      </div>

        <div class=\\"email_footer\\">
          <table class=\\"signature\\" width=\\"100%\\">
            <tr>
              <td class=\\"small_text\\" align=\\"left\\">
                Rafi Payment<sup>&#8482;</sup><br />
                <a href=\\"http://localhost:8080\\">payment.rafiproperties.com</a>
              </td>
              <td class=\\"small_text\\" align=\\"right\\">
                Customer Support: <a href=\\"mailto:leasing@rafiproperties.com\\">leasing@rafiproperties.com</a>
              </td>
            </tr>
          </table>
        </div>
      </div>
    </td>
  </tr>
</table>
</body>
</html>

  ",
  "subject": "",
}
`;
